feat(case-creation): enhance button functionality with tooltip support

tomer-shvadron opened the pull request:
- Update button state management with demo flag
- Integrate tooltips to inform users on demo account limitations
- Refactor withCaseCreation to use arrow function syntax for consistency


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced tooltips on case creation and certificate buttons to provide contextual guidance, especially for demo accounts where certain features are disabled.
  - Added a new hook for managing case creation logic, improving state management and user experience.

- **Style**
  - Updated the appearance of disabled buttons by applying reduced opacity, ensuring a more refined and consistent visual feedback.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(case-creation): enhance button functionality with tooltip support

- Update button state management with demo flag
- Integrate tooltips to inform users on demo account limitations
- Refactor withCaseCreation to use arrow function syntax for consistency

alonp99 commented on apps/backoffice-v2/src/pages/Entity/components/Case/components/CaseOptions/hooks/useCaseOptionsLogic/useCaseOptionsLogic.tsx:
```diff
@@ -22,5 +22,6 @@ export const useCaseOptionsLogic = () => {
     generateAndOpenPDFInNewTab,
     isCopyingCollectionFlowLink,
     copyCollectionFlowLink,
+    isDemo: customer?.config?.isDemo,
```
We have two flags for Demo flags, make sure the logic is activated if either flag is present

tomer-shvadron committed:
refactor(case-creation): improve demo status handling

- Update demo status determination to include demo account check
- Refactor report data processing to simplify mapping logic
